pkg/cmd/project/convert/convert.go

```diff
@@ -0,0 +1,413 @@
+type GetAllGitCredentialsCallback func() ([]*credentials.Resource, error)
+
+type ConvertFlags struct {
+	Project                    *flag.Flag[string]
+	GitUrl                     *flag.Flag[string]
+	GitBranch                  *flag.Flag[string]
+	GitCredentials             *flag.Flag[string]
+	GitUsername                *flag.Flag[string]
+	GitPassword                *flag.Flag[string]
+	GitStorage                 *flag.Flag[string]
+	GitInitialCommitMessage    *flag.Flag[string]
+	GitDefaultBranchProtected  *flag.Flag[bool]
+	GitInitialCommitBranch     *flag.Flag[string]
+	GitBasePath                *flag.Flag[string]
+	GitProtectedBranchPatterns *flag.Flag[[]string]
+}
+
+func NewConvertFlags() *ConvertFlags {
+	return &ConvertFlags{
+		Project:                    flag.New[string](FlagProject, false),
+		GitStorage:                 flag.New[string](FlagGitCredentialStorage, false),
+		GitUrl:                     flag.New[string](FlagGitUrl, false),
+		GitBranch:                  flag.New[string](FlagGitBranch, false),
+		GitInitialCommitMessage:    flag.New[string](FlagGitInitialCommitMessage, false),
+		GitCredentials:             flag.New[string](FlagGitLibraryCredentials, false),
+		GitDefaultBranchProtected:  flag.New[bool](FlagDefaultBranchProtected, false),
+		GitProtectedBranchPatterns: flag.New[[]string](FlagBranchProtectionPattern, false),
+		GitInitialCommitBranch:     flag.New[string](FlagInitialCommitBranch, false),
+		GitUsername:                flag.New[string](FlagGitUsername, false),
+		GitPassword:                flag.New[string](FlagGitPassword, true),
+		GitBasePath:                flag.New[string](FlagGitBasePath, false),
+	}
+}
+
+type ConvertOptions struct {
+	*ConvertFlags
+	*cmd.Dependencies
+	GetAllGitCredentialsCallback
+	GetAllProjectsCallback shared.GetAllProjectsCallback
+	GetProjectCallback     shared.GetProjectCallback
+}
+
+func NewConvertOptions(flags *ConvertFlags, dependencies *cmd.Dependencies) *ConvertOptions {
+	return &ConvertOptions{
+		ConvertFlags: flags,
+		Dependencies: dependencies,
+		GetAllGitCredentialsCallback: func() ([]*credentials.Resource, error) {
+			return createGetAllGitCredentialsCallback(*dependencies.Client)
+		},
+		GetProjectCallback: func(identifier string) (*projects.Project, error) {
+			return shared.GetProject(*dependencies.Client, identifier)
+		},
+		GetAllProjectsCallback: func() ([]*projects.Project, error) { return shared.GetAllProjects(*dependencies.Client) },
+	}
+}
+
+func NewCmdConvert(f factory.Factory) *cobra.Command {
+	convertProjectFlags := NewConvertFlags()
+	cmd := &cobra.Command{
+		Use:   "convert",
+		Short: "Convert a project to use Config As Code",
+		Long:  "Convert a project to use Config As Code in Octopus Deploy",
+		Example: heredoc.Docf(`
+			$ %[1]s project convert
+			$ %[1]s project convert --project "Deploy web site" --git-url https://github.com/orgname/reponame"
+		`, constants.ExecutableName),
```

Comment:
not sure if the command should be `octopus project convert` or `octopus project configure-cac` or something else

Reply:
If think `convert` is fine.